io-engine/src/lvs/lvol_snapshot.rs

```diff
@@ -792,4 +797,69 @@ impl SnapshotOps for Lvol {
             None
         }
     }
+    /// When snapshot is destroyed, reset the parent lvol usage cache and its
+    /// successor snapshot and clone usage cache.
+    fn reset_snapshot_parent_successor_usage_cache(&self) {
+        if let Some(snapshot_parent_uuid) =
+            Lvol::get_blob_xattr(self, SnapshotXattrs::ParentId.name())
+        {
+            if let Some(bdev) =
+                UntypedBdev::lookup_by_uuid_str(snapshot_parent_uuid.as_str())
+            {
+                if let Ok(parent_lvol) = Lvol::try_from(bdev) {
+                    unsafe {
+                        spdk_blob_reset_used_clusters_cache(
+                            parent_lvol.blob_checked(),
+                        );
+                    }
+                }
+            }
+            self.reset_successor_lvol_usage_cache(snapshot_parent_uuid);
+        }
+    }
+
+    /// When snapshot is destroyed, reset cache of successor snapshots and
+    /// clones based on snapshot parent uuid.
```

Comment:
Do we need to do this for the entire list or just the direct successors? Though I guess we don't have that kind of knowledge at this level?

Reply:
I think entire downstream tree need to clean as clone inherit size from snapshot.

Reply:
But I feel it is not efficient and performance is not good. Possibly need to have some enhancement in SPDK to identify the tree.